tests/queries/0_stateless/03146_parameterized_view_with_date.sql

```diff
@@ -0,0 +1,12 @@
+
+drop table if exists table_pv;
+create table table_pv (id Int32, timestamp_field DateTime) engine = Memory();
+
+insert into table_pv values(1, '2024-03-01 00:00:00');
+insert into table_pv values (2, '2024-04-01 01:00:00');
+
+create view pv as select * from table_pv where timestamp_field > {timestamp_param:DateTime};
+
+select * from pv (timestamp_param=toDateTime('2024-04-01 00:00:01'));
```

Comment:
Let's add a test for the case when result of a function is not a constant expression and expect a certain error code? Just in case to make sure that the behaviour will be expected and the user gets an understandable error